plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostingFun/Core.hs

```diff
@@ -397,17 +437,23 @@ runTwoArgumentModel
                 if size1 < size2
                     then CostLast c
                     else run (CostLast size1) (CostLast size2)
+runTwoArgumentModel
+    (ModelTwoArgumentsQuadraticInY f) =
+        lazy $ \_ costs2 ->
+            CostLast $ evaluateOneVariableQuadraticFunction f $ sumCostStream costs2
```

Comment:
This needs a comment that costing here isn't lazy, so if lazy costing is required for some future quadratic builtin, this piece of code needs to be rewritten.